frontend/src/component/demo/Demo.tsx

```diff
@@ -1,52 +1,57 @@
-import useUiConfig from 'hooks/api/getters/useUiConfig/useUiConfig';
 import { useEffect, useState } from 'react';
 import { DemoTopics } from './DemoTopics/DemoTopics';
 import { DemoSteps } from './DemoSteps/DemoSteps';
-import { ConditionallyRender } from 'component/common/ConditionallyRender/ConditionallyRender';
 import { createLocalStorage } from 'utils/createLocalStorage';
 import { TOPICS } from './demo-topics';
 import { DemoDialogWelcome } from './DemoDialog/DemoDialogWelcome/DemoDialogWelcome';
+import { DemoDialogFinish } from './DemoDialog/DemoDialogFinish/DemoDialogFinish';
 
 const defaultProgress = {
     welcomeOpen: true,
     expanded: true,
-    active: false,
-    topic: 0,
+    topic: -1,
     steps: [0],
 };
 
 const { value: storedProgress, setValue: setStoredProgress } =
     createLocalStorage('Tutorial:v1', defaultProgress);
 
 export const Demo = () => {
-    const { uiConfig } = useUiConfig();
     const [welcomeOpen, setWelcomeOpen] = useState(
         storedProgress.welcomeOpen ?? true
     );
-    const [active, setActive] = useState(false);
+    const [finishOpen, setFinishOpen] = useState(false);
+
     const [expanded, setExpanded] = useState(storedProgress.expanded ?? true);
-    const [topic, setTopic] = useState(storedProgress.topic ?? 0);
+    const [topic, setTopic] = useState(storedProgress.topic ?? -1);
```

Comment:
Would this be better?
```suggestion
    const [topic, setTopic] = useState(storedProgress.topic ?? defaultProgress.topic);
```

Reply:
If yes, the line right below might also benefit from a similar change